Missing query parameters returns HTTP 500 instead of 400

swagger endpoint definition:
```
  /dummy:
    get:
      operationId: controllers.dummy.dummy
      parameters:
        - in: query
          name: test
          type: string
          required: true
      responses:
        200:
          description: OK
          schema:
            type: object
            properties:
              data:
                type: string
```

Steps to reproduce the issue:

1. Launch app server
2. curl -I http://192.168.99.101:8080/v1/dummy --> returns HTTP 400 as expected
3. curl -I http://192.168.99.101:8080/v1/dummy?test=1 --> returns HTTP 200 as expected
4. curl -I http://192.168.99.101:8080/v1/dummy --> **returns HTTP 500**

stack trace:
```
test_1 | Traceback (most recent call last):
test_1 |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1817, in wsgi_app
test_1 |     response = self.full_dispatch_request()
test_1 |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1477, in full_dispatch_request
test_1 |     rv = self.handle_user_exception(e)
test_1 |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1381, in handle_user_exception
test_1 |     reraise(exc_type, exc_value, tb)
test_1 |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1475, in full_dispatch_request
test_1 |     rv = self.dispatch_request()
test_1 |   File "/usr/local/lib/python2.7/site-packages/flask/app.py", line 1461, in dispatch_request
test_1 |     return self.view_functions[rule.endpoint](**req.view_args)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/decorators/validation.py", line 190, in wrapper
test_1 |     response = function(*args, **kwargs)
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/decorators/produces.py", line 116, in wrapper
test_1 |     data, status_code, headers = self.get_full_response(function(*args, **kwargs))
test_1 |   File "/usr/local/lib/python2.7/site-packages/connexion/decorators/parameter.py", line 100, in wrapper
test_1 |     return function(*args, **kwargs)
test_1 | TypeError: dummy() takes exactly 1 argument (0 given)
```

@rafaelcaricio I just pushed a test against in my repo https://github.com/jvgutierrez/connexion/commit/7eba65459ad3c1bb03b7663c74df28365bc9b86f.

Check:

```
______________________________________________________________________________ test_missing_params_bug ______________________________________________________________________________

app = <connexion.app.App instance at 0x7fdda2b06c20>

    def test_missing_params_bug(app):
        app_client = app.app.test_client()
        resp = app_client.get('/v1.0/test-missing-params-bug')
        assert resp.status_code == 400
        resp = app_client.get('/v1.0/test-missing-params-bug', query_string={'p': 'test'})
        assert resp.status_code == 200
        resp = app_client.get('/v1.0/test-missing-params-bug')
>       assert resp.status_code == 400
E       assert 500 == 400
E        +  where 500 = <Response streamed [500 INTERNAL SERVER ERROR]>.status_code
```

--- a/connexion/decorators/validation.py
+++ b/connexion/decorators/validation.py
@@ -11,6 +11,7 @@ Unless required by applicable law or agreed to in writing, software distributed 
  language governing permissions and limitations under the License.
 """
 
+import copy
 import flask
 import functools
 import itertools
@@ -133,6 +134,7 @@ class ParameterValidator():
             except TypeValidationError as e:
                 return str(e)
 
+            param = copy.deepcopy(param)
             if 'required' in param:
                 del param['required']
             try:

--- a/tests/fakeapi/api.yaml
+++ b/tests/fakeapi/api.yaml
@@ -646,6 +646,16 @@ paths:
           in: query
           default: false
 
+  /test-required-param:
+    get:
+      summary: Test required param without default value
+      operationId: fakeapi.hello.test_required_param
+      parameters:
+        - name: simple
+          type: string
+          in: query
+          required: true
+
 definitions:
   new_stack:
     type: object

--- a/tests/fakeapi/hello.py
+++ b/tests/fakeapi/hello.py
@@ -230,3 +230,7 @@ def test_formData_missing_param():
 
 def test_bool_default_param(thruthiness):
     return thruthiness
+
+
+def test_required_param(simple):
+    return simple

--- a/tests/test_app.py
+++ b/tests/test_app.py
@@ -600,3 +600,16 @@ def test_bool_as_default_param(app):
     assert resp.status_code == 200
     response = json.loads(resp.data.decode())
     assert response == True
+
+
+def test_required_param_miss_config(app):
+    app_client = app.app.test_client()
+
+    resp = app_client.get('/v1.0/test-required-param')
+    assert resp.status_code == 400
+
+    resp = app_client.get('/v1.0/test-required-param', query_string={'simple': 'test'})
+    assert resp.status_code == 200
+
+    resp = app_client.get('/v1.0/test-required-param')
+    assert resp.status_code == 400

